lib/rules/no-warning-comments.js

```diff
@@ -135,18 +144,39 @@ module.exports = {
          * @returns {void} undefined.
          */
         function checkComment(node) {
-            if (astUtils.isDirectiveComment(node) && selfConfigRegEx.test(node.value)) {
+            if (
+                astUtils.isDirectiveComment(node) &&
+                selfConfigRegEx.test(node.value)
+            ) {
                 return;
             }
 
             const matches = commentContainsWarningTerm(node.value);
 
-            matches.forEach(matchedTerm => {
+            matches.forEach(({ type, comment }) => {
+                let commentToDisplay = "";
+                let truncated = false;
+
+                for (const c of comment.trim().split(/\s+/u)) {
+                    commentToDisplay = commentToDisplay.trim();
+                    const tmp = `${commentToDisplay} ${c}`;
```

Comment:
This will have one extra space before the first word (at least in the first iteration), so it could make the check bellow to be off-by-one.

Can we maybe do something like this:

```js
const tmp = commentToDisplay ? `${commentToDisplay} ${c}` : c;
```